src/drivers/intel/dmic.c

```diff
@@ -1529,7 +1529,7 @@ static int dmic_trigger(struct dai *dai, int cmd, int direction)
 		break;
 	case COMP_TRIGGER_PAUSE:
 		dmic->state = COMP_STATE_PAUSED;
-		dmic_stop(dai, true);
+		dmic_stop(dai, false);
```

Comment:
@kv2019i is this a better fix?
```
if (dmic->state == COMP_STATE_PREPARE || dmic->state == COMP_STATE_PAUSED)		 
   (*uncached_dmic_active_fifos)++;
```

Reply:
@kv2019i I'm worried about active_fifos_count not tracking stopped FIFOs HW even if it fixes the reference loss. Non-matching count will alter the stop sequence and could leave bus clock on. I'm not sure about and need to test this.

 @ranj063 Wouldn't that remove @lyakh 's add? Instead of in_active flag I wonder if the active fifos count should be replaced with a bit-field. In dmic_stop() the corresponding bit to DAI # is cleared, and set it dmic_start().

Reply:
@ranj063 We shouldn't increase fifo-count when state is paused, so that's not right. I'd vote for using a bitfield instead. This would be both easier to understand and also easier to debug than having a counter.